src/main/java/org/unice/service/CustomerService.java

```diff
@@ -70,9 +71,19 @@ public static void delete(String id){
      * Create the index used by redis search
      */
     private static void createIndexIfNotExists(){
+        boolean indexExists = false;
+
         try {
-            database.ftInfo(indexName);
+            var info = database.ftInfo(indexName);
+
+            if (info != null && info.size() > 0) {
+                indexExists = true;
+            }
         } catch (JedisDataException exception) {
+            // Ignore the exception if the index does not exist
```

Comment:
I would just add here maybe System.out.println("This index already exists")

Reply:
No, it's the opposite. An error is throw when the index doesn't exist. I don't want to print message because the method is call too many time when i try to get data.

Reply:
for instance, if i get 100 customer in a loop using getById, it will print "This index already exists" * 100